src/Task.js

```diff
@@ -0,0 +1,46 @@
+import './Task.css'
+import {AppModes} from "./App";
+
+function Task(props) {
+    return (
+        <li>
+        {props.appMode === AppModes.EDIT_MODE ?
```

Comment:
Same suggestion for huge ?:

Reply:
There's a lot of duplication between the ? and : cases.  I'd extract that into common code.

Reply:
In new commits, we reworked this to not actually use a giant ternary statement. Looking forward to hearing what you think, and thanks for all the feedback!